### ISSUE TITLE
Partition leads to invalid object identifier error [KaHyPar v1.3.0]

### ISSUE THREAD

**Issue by jalving** [kahypar/KaHyPar.jl#24]
The Julia partition function fails when using KaHyPar v1.3.0. The following code produces the failure:

```julia
using KaHyPar
using SparseArrays

I = [1,3,1,2,4,5,4,5,7,3,6,7]
J = [1,1,2,2,2,2,3,3,3,4,4,4]
V = Int.(ones(length(I)))

A = sparse(I,J,V)

h = KaHyPar.HyperGraph(A)

KaHyPar.partition(h,2,configuration = :edge_cut)
```

Here is the output:

```
+++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++ 
+                    _  __     _   _       ____                               + 
+                   | |/ /__ _| | | |_   _|  _ \ __ _ _ __                    + 
+                   | ' // _` | |_| | | | | |_) / _` | '__|                   + 
+                   | . \ (_| |  _  | |_| |  __/ (_| | |                      + 
+                   |_|\_\__,_|_| |_|\__, |_|   \__,_|_|                      + 
+                                    |___/                                    + 
+                 Karlsruhe Hypergraph Partitioning Framework                 + 
+++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++ 
*******************************************************************************
*                            Partitioning Context                             *
*******************************************************************************
Partitioning Parameters:
  Hypergraph:                         
  Partition File:                     
  Mode:                               direct
  Objective:                          cut
  k:                                  2
  epsilon:                            0.03
  seed:                               -1
  # V-cycles:                         0
  time limit:                         -1s
  hyperedge size threshold:           1000
  use individual block weights:       false
  L_opt:                              4
  L_max:                              4
-------------------------------------------------------------------------------
Preprocessing Parameters:
  enable deduplication:               false
  enable min hash sparsifier:         true
  enable community detection:         true
-------------------------------------------------------------------------------
MinHash Sparsifier Parameters:
  max hyperedge size:                 1200
  max cluster size:                   10
  min cluster size:                   2
  number of hash functions:           5
  number of combined hash functions:  100
  active at median net size >=:       28
  sparsifier is active:               false
-------------------------------------------------------------------------------
Community Detection Parameters:
  use community detection in IP:      true
  maximum louvain-pass iterations:    100
  minimum quality improvement:        0.0001
  graph edge weight:                  degree
  reuse community structure:          false
-------------------------------------------------------------------------------
Coarsening Parameters:
  Algorithm:                          ml_style
  max-allowed-weight-multiplier:      1
  contraction-limit-multiplier:       160
  hypernode weight fraction:          0.003125
  max. allowed hypernode weight:      1
  contraction limit:                  320
  Rating Parameters:
    Rating Function:                  heavy_edge
    Use Community Structure:          true
    Heavy Node Penalty:               no_penalty
    Acceptance Policy:                best_prefer_unmatched
    Partition Policy:                 normal
    Fixed Vertex Acceptance Policy:   fixed_vertex_allowed
-------------------------------------------------------------------------------
Initial Partitioning Parameters:
  # IP trials:                        20
  Mode:                               recursive
  Technique:                          multilevel
  Algorithm:                          pool
  Bin Packing algorithm:              UNDEFINED
    early restart on infeasible:      false
    late restart on infeasible:       false
IP Coarsening:                        
Coarsening Parameters:
  Algorithm:                          ml_style
  max-allowed-weight-multiplier:      1
  contraction-limit-multiplier:       150
  hypernode weight fraction:          determined before IP
  max. allowed hypernode weight:      determined before IP
  contraction limit:                  determined before IP
  Rating Parameters:
    Rating Function:                  heavy_edge
    Use Community Structure:          true
    Heavy Node Penalty:               no_penalty
    Acceptance Policy:                best_prefer_unmatched
    Partition Policy:                 normal
    Fixed Vertex Acceptance Policy:   fixed_vertex_allowed
IP Local Search:                      
Local Search Parameters:
  Algorithm:                          twoway_fm
  iterations per level:               2147483647
  stopping rule:                      simple
  max. # fruitless moves:             50
  Flow Refinement Parameters:
    execution policy:                 UNDEFINED
-------------------------------------------------------------------------------
Local Search Parameters:
  Algorithm:                          kway_fm_hyperflow_cutter
  iterations per level:               2147483647
  stopping rule:                      adaptive_opt
  adaptive stopping alpha:            1
  Flow Refinement Parameters:
    execution policy:                 exponential
-------------------------------------------------------------------------------
 
Invalid object identifier 
```

@SebastianSchlag do you immediately notice any issues in the output? Did any of the C interface functions change for v1.3.0?

**Comment by jalving** [comment]
Linking latest build log [here](https://dev.azure.com/JuliaPackaging/Yggdrasil/_build/results?buildId=23049&view=logs&jobId=3edbdc5e-871f-59b3-17c4-94311d52c9e1&j=979f698a-bb1e-5377-f84f-ad70f66e135d&t=2da6b2da-9784-5d8a-1f8d-782a3dbd72ca). I am doing some tests with the packaged binaries in the meantime.

**Comment by jalving** [comment]
The problem turns out to be out-of-date config files packaged with the Julia interface. My guess is that KaHyPar does not apply default values if none are given? The fix is to simply copy the latest config files to this package.

**Comment by SebastianSchlag** [comment]
> The problem turns out to be out-of-date config files packaged with the Julia interface. My guess is that KaHyPar does not apply default values if none are given? The fix is to simply copy the latest config files to this package.

Thanks for investigating this, @jalving. Yeah, KaHyPar intentionally does not use any default values to really make sure it captures the intended user behavior.

**Comment by jalving** [comment]
should be fixed by PR #25

**Comment by SebastianSchlag** [comment]
Follow-up to get more meaningful error message in this case: https://github.com/kahypar/kahypar/pull/131
